node/src/lib.rs

```diff
@@ -90,6 +92,15 @@ pub trait Network: Send + Sync + 'static {
 
     /// Retrieves number of alive nodes
     async fn alive_nodes_count(&self) -> usize;
+
+    async fn wait_for_alive_nodes(&self, amount: usize, retry: usize) {
+        let mut retry = retry;
+        while self.alive_nodes_count().await < amount && retry > 0 {
+            warn!("wait_for_alive_nodes");
+            tokio::time::sleep(Duration::from_secs(2)).await;
+            retry -= 1;
+        }
+    }
```

Comment:
I think it would make more sense for `wait_alive_nodes` to be an abstract method of the Network trait and let the implementor define its behaviour? Also, it would make much more sense to include a timeout as a parameter instead of a retry. Something like:

```rust
pub enum AlivePeerResult {
    Ready,
    TimedOut(usize), // the usize being the number of alive nodes when timed out
    Error(Box<dyn Error + Send + Sync>),
}

async fn wait_for_alive_nodes(&self, amount: usize, timeout: Duration) -> AlivePeerResult;
```

Reply:
> I think it would make more sense for wait_alive_nodes to be an abstract method of the Network trait and let the implementor define its behaviour?

`wait_alive_nodes` is indeed a method of the network trait, it has a default implementation that the implementor can override if needed

>  would make much more sense to include a timeout as a parameter instead of a retry

Having a timeout would still need an internal retry/polling mechanism, so I'm not sure what would be the proper implementation. Do you mind to open a techdebt followup issue to discuss?